server/startup/avatar.coffee

```diff
@@ -43,6 +43,16 @@ Meteor.startup ->
 		res.setHeader 'Content-Disposition', 'inline'
 
 		if not file?
+
+			if Meteor.settings?.public?.sandstorm
+				user = RocketChat.models.Users.findOneByUsername(params.username.replace('.jpg', ''))
+				if user?.services?.sandstorm?.picture
+					console.log "Should redirect to: ", user.services.sandstorm.picture
+					res.setHeader 'Location', user.services.sandstorm.picture
+					res.writeHead 302
```

Comment:
This will work, but as mentioned on the bug thread I would recommend setting a non-zero cache max-age so that the browser doesn't have to repeatedly incur a network round trip to fetch their avatar every time the user says something in chat. However, a non-zero value means that avatar changes won't take effect immediately. If the redirect could be avoided by embedding the final URL directly on the client side, it would both perform well and correctly update on-demand (reactively, even!). Any possibility of doing it that way?